trie/stacktrie.go

```diff
@@ -27,6 +27,7 @@ import (
 
 var (
 	stPool = sync.Pool{New: func() any { return new(stNode) }}
+	bPool  = newBytesPool(32, 100)
```

Comment:
any particular reason to not use `sync.Pool`?

Reply:
Yes! You'd be surprised, but the whole [problem I had with extra alloc](https://github.com/ethereum/go-ethereum/pull/30743#issuecomment-2469038633) : 

![Screenshot 2024-11-11 at 22-18-40 trie test alloc_space](https://github.com/user-attachments/assets/522ab0fa-08a3-46b0-a41a-cd132edd4c1c)

I solved that by not using a `sync.Pool`. I suspect it's due to the interface-conversion, but I don't know any deeper details about the why of it.